include/iterlib/FilterIterator.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include "iterlib/WrappedIterator.h"
+
+namespace iterlib {
+
+enum class FilterType {
```

Comment:
Maybe we can add some logical separation between the parts of code that handle single value filter and multivalue filtering. It might make reading the code easier.